templates/download_source

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash -eu
+#
+# Copyright (C) 2017 Google Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+readonly BLACKLIST_FILE="/tmp/blacklisted_packages.list"
+
+function list_all_packages() {
+  dpkg-query -W | cut -f 1 | cut -d ":" -f 1 | sort
+}
+
+function list_packages() {
+  comm -13 <(sort "${BLACKLIST_FILE}") <(list_all_packages)
+}
+
+function list_gpl_packages() {
+  for pkg in $(list_packages)
+  do
+    if egrep -q '\WGPL\W' "/usr/share/doc/${pkg}/copyright"
+    then
+      echo "${pkg}"
+    fi
+  done
+}
+
+function blacklist_all_current() {
+  list_all_packages >> "${BLACKLIST_FILE}"
+}
+
+function blacklist_nothing() {
+  echo -n > "${BLACKLIST_FILE}"
```

Comment:
This seems be a little more clear to me:

`cat /dev/null > "${BLACKLIST_FILE}"`